test/AllToAllVTests.cpp

```diff
@@ -137,13 +155,31 @@ namespace RcclUnitTesting
     TestBed testBed;
 
     // Configuration
-    std::vector<ncclDataType_t> const& dataTypes       = {ncclFloat32, ncclInt8};
+    std::vector<ncclDataType_t> const& testDataTypes   = {ncclFloat32, ncclInt8};
     bool                        const  inPlace         = false;
     bool                        const  useManagedMem   = false;
     bool                        const  useHipGraph     = false;
 
     OptionalColArgs options;
 
+    // Filter out any unsupported datatypes, in case only subset has been compiled for
+    std::vector<ncclDataType_t> const& supportedDataTypes = testBed.GetAllSupportedDataTypes();
+    std::vector<ncclDataType_t> dataTypes;
+    for (auto dt : testDataTypes)
+    {
+      for (int i = 0; i < supportedDataTypes.size(); ++i)
+      {
+        if (supportedDataTypes[i] == dt)
+        {
+          dataTypes.push_back(dt);
+          break;
+        }
+      }
+    }
```

Comment:
This can be moved to a static util function and reused across tests. Same for supported ops